distribution/src/config/jvm.options

```diff
@@ -58,6 +58,10 @@
 # result in less optimal vector performance
 20-:--add-modules=jdk.incubator.vector
 
+# Required to workaround perform issue in JDK 23, https://github.com/elastic/elasticsearch/issues/113030
+23-23:-XX:CompileCommand=dontinline,java/lang/invoke/MethodHandle.setAsTypeCache
```

Comment:
nit: the jvm options format allows for simply specifying the major version, no need for the range to itself